### PR TITLE
use new interface to create threads on HermitCore

### PR DESCRIPTION
**Pull request by stlankes**
- the new interface allows to define the stack size
- increase the default stack size to 1 MByte

### CODE REVIEW

File: src/libstd/sys/hermit/thread.rs [hunk 1]
```diff
@@ -16,25 +16,24 @@ pub struct Thread {
 unsafe impl Send for Thread {}
 unsafe impl Sync for Thread {}
 
-pub const DEFAULT_MIN_STACK_SIZE: usize = 262144;
+pub const DEFAULT_MIN_STACK_SIZE: usize = 1_048_576;
```
Comment (lzutao) [inline src/libstd/sys/hermit/thread.rs:19]:
```suggestion
pub const DEFAULT_MIN_STACK_SIZE: usize = 1 << 20;
```

Comment (stlankes) [reply src/libstd/sys/hermit/thread.rs]: You are right. Ich changed the line.

**Comment by Dylan-DPC** [comment]
@hanna-kruppe waiting for your review on this :)

**Pull request merged**